image.go

```diff
@@ -321,10 +321,7 @@ func (i *containerImageRef) NewImageSource(ctx context.Context, sc *types.System
 		}
 		// If we're not re-exporting the data, and we're reusing layers individually, reuse
 		// the blobsum and diff IDs.
-		if !i.exporting && !i.squash && layerID != i.layerID {
-			if layer.UncompressedDigest == "" {
-				return nil, errors.Errorf("unable to look up size of layer %q", layerID)
-			}
+		if !i.exporting && !i.squash && layerID != i.layerID && layer.UncompressedDigest != "" {
```

Comment:
I'm going back and forth.  I'm just wondering if we should keep the if statement still, but replace the return with a debug.  I'm leaning towards no, but not fully  @nalind thoughts?

Reply:
Well right now, we don't set this information on the mapped layer in the storage library, and this forces us to reconstruct the layer in such a way that its diffID might be different than that of the layer that it was based off of.  We either do that, or we try to redirect ourselves to the corresponding not-mapped layer, and I don't think the storage library gives us quite enough information to do that.